### ISSUE TITLE
usb.core.USBError: [Errno None] b'libusb0-dll:err [_usb_reap_async] timeout error on Samsung

### ISSUE THREAD

**Issue by iman432** [fgsect/scat#55]
Hi all,

someone knows what is issue with this error ?

 .\python.exe .\scat-master\scat.py -t qc -u -a 000:255 -i 0 --qmdl output.qmdl -F sim1.pcap
Trying USB device at address 000:255
  CONFIGURATION 1: 500 mA ==================================
   bLength              :    0x9 (9 bytes)
   bDescriptorType      :    0x2 Configuration
   wTotalLength         :   0x27 (39 bytes)
   bNumInterfaces       :    0x1
   bConfigurationValue  :    0x1
   iConfiguration       :    0x4 rmnet_acm_diag_adb
   bmAttributes         :   0x80 Bus Powered
   bMaxPower            :   0xfa (500 mA)
    INTERFACE 4: Vendor Specific ===========================
     bLength            :    0x9 (9 bytes)
     bDescriptorType    :    0x4 Interface
     bInterfaceNumber   :    0x4
     bAlternateSetting  :    0x0
     bNumEndpoints      :    0x3
     bInterfaceClass    :   0xff Vendor Specific
     bInterfaceSubClass :   0xff
     bInterfaceProtocol :   0x50
     iInterface         :    0x9 RmNet
      ENDPOINT 0x85: Interrupt IN ==========================
       bLength          :    0x7 (7 bytes)
       bDescriptorType  :    0x5 Endpoint
       bEndpointAddress :   0x85 IN
       bmAttributes     :    0x3 Interrupt
       wMaxPacketSize   :    0x8 (8 bytes)
       bInterval        :    0x9
      ENDPOINT 0x8E: Bulk IN ===============================
       bLength          :    0x7 (7 bytes)
       bDescriptorType  :    0x5 Endpoint
       bEndpointAddress :   0x8e IN
       bmAttributes     :    0x2 Bulk
       wMaxPacketSize   :  0x200 (512 bytes)
       bInterval        :    0x0
      ENDPOINT 0xF: Bulk OUT ===============================
       bLength          :    0x7 (7 bytes)
       bDescriptorType  :    0x5 Endpoint
       bEndpointAddress :    0xf OUT
       bmAttributes     :    0x2 Bulk
       wMaxPacketSize   :  0x200 (512 bytes)
       bInterval        :    0x0
2022-02-16 17:31:35,615 scat.qualcommparser (stop_diag) INFO: Stopping diag
Traceback (most recent call last):
  File "C:\Users\i84153596\AppData\Local\Programs\Python\Python39\scat-master\scat.py", line 158, in <module>
    current_parser.stop_diag()
  File "C:\Users\i84153596\AppData\Local\Programs\Python\Python39\scat-master\parsers\qualcomm\qualcommparser.py", line 236, in stop_diag
    self.io_device.write_then_read_discard(util.generate_packet(struct.pack('<BB', diagcmd.DIAG_EVENT_REPORT_F, 0x00)), 0x1000, False)
  File "C:\Users\i84153596\AppData\Local\Programs\Python\Python39\scat-master\iodevices\usbio.py", line 32, in write_then_read_discard
    self.write(write_buf, encode_hdlc)
  File "C:\Users\i84153596\AppData\Local\Programs\Python\Python39\scat-master\iodevices\usbio.py", line 29, in write
    self.w_handle.write(write_buf)
  File "C:\Users\i84153596\AppData\Local\Programs\Python\Python39\lib\site-packages\usb\core.py", line 408, in write
    return self.device.write(self, data, timeout)
  File "C:\Users\i84153596\AppData\Local\Programs\Python\Python39\lib\site-packages\usb\core.py", line 989, in write
    return fn(
  File "C:\Users\i84153596\AppData\Local\Programs\Python\Python39\lib\site-packages\usb\backend\libusb0.py", line 545, in bulk_write
    return self.__write(_lib.usb_bulk_write,
  File "C:\Users\i84153596\AppData\Local\Programs\Python\Python39\lib\site-packages\usb\backend\libusb0.py", line 681, in __write
    return int(_check(fn(
  File "C:\Users\i84153596\AppData\Local\Programs\Python\Python39\lib\site-packages\usb\backend\libusb0.py", line 447, in _check
    raise USBError(errmsg, ret)
usb.core.USBError: [Errno None] b'libusb0-dll:err [_usb_reap_async] timeout error\n'

**Comment by Cainor** [comment]
I'm really not an expert here .. but I tried the following to overcome all errors:
1- Choose "DM + ACM + ADB" from the `*#0808#` menu.
2- Reboot the device.
3- Make sure if you have Qualcomm or Exynos processer to choose the right `-t` mode.

I have a Samsung S10+ Exynos version and here is my final command:
`python3 scat.py -t sec -m e303 -u -a 002:019 -i 2`

However, even though I got no errors, seems like I'm not capturing any thing at all. Here is what is showing in the terminal:

`
... Snap ...
       bInterval        :    0x0
      ENDPOINT 0x84: Bulk IN ===============================
       bLength          :    0x7 (7 bytes)
       bDescriptorType  :    0x5 Endpoint
       bEndpointAddress :   0x84 IN
       bmAttributes     :    0x2 Bulk
       wMaxPacketSize   :  0x200 (512 bytes)
       bInterval        :    0x0
2022-03-01 02:31:06,528 scat.samsungparser (stop_diag) INFO: Stopping diag
2022-03-01 02:31:06,531 scat.samsungparser (init_diag) INFO: Initialize diag
2022-03-01 02:31:06,598 scat.samsungparser (run_diag) INFO: Starting diag

`

And that is it.
